core/src/layers/foyer.rs

```diff
@@ -0,0 +1,420 @@
+        let r = args.range();
+        let path = path.to_string();
+        async move {
+            let entry = self
+                .inner
+                .cache
+                .fetch(path.clone(), || {
+                    let inner = self.inner.clone();
+                    async move {
+                        let (_, mut reader) = inner
+                            .accessor
+                            .read(&path, args.with_range(BytesRange::new(0, None)))
+                            .await
+                            .map_err(FoyerError::other)?;
+                        let buffer = reader.read_all().await.map_err(FoyerError::other)?;
+                        Ok(buffer.into())
+                    }
+                })
+                .await
+                .map_err(extract_err)?;
+
+            let r = r.to_range();
+            let start = match r.start_bound() {
+                Bound::Included(i) => *i,
+                Bound::Excluded(i) => *i + 1,
+                Bound::Unbounded => 0,
+            };
+            let end = match r.end_bound() {
+                Bound::Included(i) => *i + 1,
+                Bound::Excluded(i) => *i,
+                Bound::Unbounded => entry.len() as u64,
+            };
+            let range = BytesContentRange::default()
+                .with_range(start, end - 1)
+                .with_size(entry.len() as _);
+            let buffer = entry.slice(start as usize..end as usize);
+            let rp = RpRead::new()
+                .with_size(Some(buffer.len() as _))
+                .with_range(Some(range));
+            Ok((rp, buffer))
+        }
+    }
+
+    fn write(
+        &self,
+        path: &str,
+        args: OpWrite,
+    ) -> impl Future<Output = Result<(RpWrite, Self::Writer)>> + MaybeSend {
+        let inner = self.inner.clone();
+        async move {
+            let (rp, w) = self.inner.accessor.write(path, args).await?;
+            Ok((
+                rp,
+                Writer {
+                    w,
+                    q: Some(QueueBuf::new()),
+                    path: path.to_string(),
+                    inner,
+                },
+            ))
+        }
+    }
+
+    fn delete(&self) -> impl Future<Output = Result<(RpDelete, Self::Deleter)>> + MaybeSend {
+        let inner = self.inner.clone();
+        async move {
+            let (rp, d) = inner.accessor.delete().await?;
+            Ok((
+                rp,
+                Deleter {
+                    d,
+                    keys: vec![],
+                    inner,
+                },
+            ))
+        }
+    }
+
+    fn list(
+        &self,
+        path: &str,
+        args: OpList,
+    ) -> impl Future<Output = Result<(RpList, Self::Lister)>> + MaybeSend {
+        self.inner.accessor.list(path, args)
+    }
+
+    fn info(&self) -> std::sync::Arc<AccessorInfo> {
+        self.inner.accessor.info()
+    }
+
+    fn create_dir(
```

Comment:
We don't need to implement this since we already have our auto-forward implementation.